hazelcast/src/test/java/com/hazelcast/spi/discovery/DiscoverySpiTest.java

```diff
@@ -110,7 +113,66 @@ public void whenStrategyClassNameNotExist_thenFailFast() {
         config.getNetworkConfig().getJoin().getMulticastConfig().setEnabled(false);
 
         createHazelcastInstance(config);
+    }
+
+    @Test
+    public void givenDiscoveryStrategyFactoryExistOnClassPath_whenTheSameFactoryIsConfiguredExplicitly_thenOnlyOneInstanceOfStrategyIsCreated() {
```

Comment:
longest methodname in the system?

Reply:
The longest one should be: `newMethodGetter_whenExtractingFromNonEmpty_Collection_bothNullFirst_FieldAndParentIsNonEmptyMultiResult_nullValueFirst_thenInferReturnType()` (so no new winner)
```bash
egrep -R -ho "public void (.*)\(\)" * | awk '{ print length($0) " " $0; }' $file | sort -r -n | cut -d ' ' -f 2-
```